src/aead/mod.rs

```diff
@@ -338,3 +342,118 @@ fn check_per_nonce_max_bytes(in_out_len: usize)
     }
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use std::iter;
+    use std::vec::Vec;
+
+    use test;
+    use super::*;
+
+    #[test]
+    fn test_aes_128_gcm_truncated_tag_96() {
```

Comment:
In aead_test.rs, there are tests `aead_aes_gcm_128()`, `aead_aes_gcm_256()`, and `aead_chacha20_poly1305()`. Please move your tests there and make them work the exact same way as those existing tests. In particular, you should use the `test_aead()` function. Also move chacha_poly_truncated_tag_96.txt and aes_gcm_truncated_tag_96.txt to that same directory.

Note in particular that `test_aead()` uses the same inputs to test both encryption and decryption so you don't need separate [encrypt] and [decrypt] sections in your test files; just make them all one big section.

The tests are in tests/aead_test.rs because they're testing the public API; we generally put tests of the public API in tests under tests/ instead of with these kind of same-module-file unit tests.